### PR TITLE
Remove syn-related API

### PR DESCRIPTION
**Pull request by scrabsha**

### CODE REVIEW

File: src/diagnosis.rs [hunk 1]
```diff
@@ -29,53 +22,72 @@ impl DiagnosisCollector {
...
     }
 }
 
 #[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub(crate) struct DiagnosisItem {
     kind: DiagnosisItemKind,
-    path: ItemPath,
-    trait_impl: Option<Ident>,
+    path: String,
+    trait_impl: Option<String>,
```
Comment (o0Ignition0o) [inline src/diagnosis.rs:66]:
```suggestion
```

File: src/diagnosis.rs [hunk 2]
```diff
@@ -29,53 +22,72 @@ impl DiagnosisCollector {
...
 }
 
 impl DiagnosisItem {
-    pub(crate) fn removal(path: ItemPath, trait_impl: Option<Ident>) -> DiagnosisItem {
+    pub(crate) fn removal(path: String) -> DiagnosisItem {
         DiagnosisItem {
             kind: DiagnosisItemKind::Removal,
             path,
-            trait_impl,
+            trait_impl: None,
```
Comment (o0Ignition0o) [inline src/diagnosis.rs:74]:
```suggestion
```

File: src/diagnosis.rs [hunk 3]
```diff
@@ -29,53 +22,72 @@ impl DiagnosisCollector {
...
         }
     }
 
-    pub(crate) fn modification(path: ItemPath, trait_impl: Option<Ident>) -> DiagnosisItem {
+    pub(crate) fn modification(path: String) -> DiagnosisItem {
         DiagnosisItem {
             kind: DiagnosisItemKind::Modification,
             path,
-            trait_impl,
+            trait_impl: None,
```
Comment (o0Ignition0o) [inline src/diagnosis.rs:82]:
```suggestion
```

File: src/diagnosis.rs [hunk 4]
```diff
@@ -29,53 +22,72 @@ impl DiagnosisCollector {
...
         }
     }
 
-    pub(crate) fn addition(path: ItemPath, trait_impl: Option<Ident>) -> DiagnosisItem {
+    pub(crate) fn addition(path: String) -> DiagnosisItem {
         DiagnosisItem {
             kind: DiagnosisItemKind::Addition,
             path,
-            trait_impl,
+            trait_impl: None,
```
Comment (o0Ignition0o) [inline src/diagnosis.rs:90]:
```suggestion
```

File: src/glue.rs [hunk 5]
```diff
@@ -5,71 +5,139 @@ use std::{
...
-    fn fmt(&self, f: &mut Formatter) -> FmtResult {
-        write!(f, "rustc yielded an invalid program: {}", self.0)
+        Compilation::Stop
     }
 }
 
-impl Error for InvalidRustcAst {}
+fn get_diagnosis(tcx: &TyCtxt) -> AnyResult<ApiCompatibilityDiagnostics> {
+    let (prev, curr) = get_crates_id(tcx).context("Failed to get dependencies crate id")?;
+    let comparator = compare(tcx, prev, curr);
```
Comment (o0Ignition0o) [inline src/glue.rs:103]: 👀

File: src/glue.rs [hunk 6]
```diff
@@ -5,71 +5,139 @@ use std::{
...
+
+    Ok(comparator.run(tcx))
+}
+
+pub fn get_crates_id(tcx: &TyCtxt) -> AnyResult<(CrateNum, CrateNum)> {
```
Comment (o0Ignition0o) [inline src/glue.rs:108]: probably not the best name here 😅

File: src/glue.rs [hunk 7]
```diff
@@ -5,71 +5,139 @@ use std::{
...
+            ExternCrateSource::Path => return false,
+        }
+    } else {
+        false
+    }
+}
 
-    let prev_api = PublicApi::from_ast(&prev_ast);
-    let curr_api = PublicApi::from_ast(&curr_ast);
+pub fn compare(tcx: &TyCtxt, prev: CrateNum, curr: CrateNum) -> ApiComparator {
```
Comment (o0Ignition0o) [inline src/glue.rs:138]: ditto 😅

**Review by o0Ignition0o — APPROVED** [review]

File: src/diagnosis.rs (hunk 1 above)
Comment (scrabsha) [reply src/diagnosis.rs]: Fixed by [d0847c4](https://github.com/iomentum/cargo-breaking/pull/24/commits/d0847c4cbadaa8c6d33c062c4a67dc476d9e3a74).

**Pull request merged**